gdk4/src/texture.rs

```diff
@@ -18,3 +21,51 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub fn for_pixbuf_future(
+        pixbuf: &'static gdk_pixbuf::Pixbuf,
+    ) -> Pin<Box<dyn Future<Output = Option<Texture>> + 'static>> {
+        Box::pin(async move {
+            let (sender, receiver) = futures_channel::oneshot::channel();
+
+            let sender = std::cell::Cell::new(Some(sender));
+
+            let task = gio::Task::new(
+                Some(&pixbuf.upcast_ref()),
+                gio::Cancellable::NONE,
+                move |result, _| {
+                    let texture = result
+                        .downcast_ref::<gio::Task>()
```

Comment:
@pbor This is always a `gio::Task` or not? Can we update the closure signature maybe?

Reply:
I guess... I seem to recall the reason why the C code uses `GAsyncResult` was mostly historical?